server/current-context.js

```diff
@@ -78,7 +78,28 @@ LoopBackContext.createContext = function(scopeName) {
     process.context[scopeName] = ns;
     // Set up LoopBackContext.getCurrentContext()
     LoopBackContext.getCurrentContext = function() {
-      return ns && ns.active ? ns : null;
+      if (ns && ns.active) {
+        /**
+         * **NOTE**
+         * This only re-binds get and set methods, the most used.
+         * If you use other methods of the context, e.g. runInContext(), etc.,
+         * you may run into unexpected issues that are fixed only for get, set.
+         */
+        var boundMethods = {
+          get: ns.bind(ns.get),
+          set: ns.bind(ns.set),
+        };
+        var handler = {
+          get: function(target, name) {
+            return ['get', 'set'].includes(name) ?
+            boundMethods[name] :
+            target[name];
+          },
+        };
+        return new Proxy(ns, handler);
```

Comment:
I find using `Proxy` rather heavy weight. How about using regular prototypal inheritance, would it not work equally well?

```js
var boundContext = Object.create(ns);
boundContext.get = ns.bind(ns.get);
boundContext.set = ns.bind(ns.set);
return boundContext;
```